pydm/tests/widgets/test_slider.py

```diff
@@ -0,0 +1,659 @@
+    Given the user's min and max values, and a value to move the slider to, the new position for the slider must be
+    correctly calculated.
+
+    Parameters
+    ----------
+    qtbot : fixture
+        Window for widget testing
+    new_value : int
+        The new value for the widget
+    expected_slider_value : int
+        The new calculcated widget value
+    """
+    pydm_slider = PyDMSlider()
+    qtbot.addWidget(pydm_slider)
+
+    pydm_slider.userDefinedLimits = True
+    pydm_slider.userMinimum = minimum
+    pydm_slider.userMaximum = maximum
+
+    pydm_slider._slider.setValue(0)
+    assert pydm_slider._slider.value() == 0
+
+    expected_slider_value = np.argmin(abs(pydm_slider._slider_position_to_value_map - float(new_value)))
+    pydm_slider.set_slider_to_closest_value(new_value)
+
+    if new_value is None or pydm_slider._needs_limit_info:
+        assert pydm_slider._silder.value() == 0
+    else:
+        assert pydm_slider._mute_internal_slider_changes is False
+        assert pydm_slider._slider.value() == expected_slider_value
+
+
+@pytest.mark.parametrize("new_channel_value, is_slider_down", [
+    (15, False),
+    (15, True),
+])
+def test_value_changed(qtbot, signals, monkeypatch, new_channel_value, is_slider_down):
+    """
+    Test the updating of the widget's slider component value when the channel value has changed.
+
+    Expectations:
+    The widget's text component will display the correct new value, and the widget's slider component will reflect
+    the right movement as calculated.
+
+    Parameters
+    ----------
+    qtbot : fixture
+        Window for widget testing
+    signals : fixture
+        The signals fixture, which provides access signals to be bound to the appropriate slots
+    monkeypatch : fixture
+        To override the default behavior of isSliderDown while simulating whether the widget's slider is being held down
+        by the user or not
+    new_channel_value : int
+        The new value coming from the channel
+    is_slider_down : bool
+        True if the slider is to be simulated as being held down by the user; False otherwise.
+    """
+    pydm_slider = PyDMSlider()
+    qtbot.addWidget(pydm_slider)
+
+    pydm_slider.userDefinedLimits = True
+    pydm_slider.userMinimum = 10
+    pydm_slider.userMaximum = 100
+
+    pydm_slider._slider.setValue(0)
+    assert pydm_slider._slider.value() == 0
+
+    monkeypatch.setattr(QSlider, "isSliderDown", lambda *args: is_slider_down)
+    signals.new_value_signal.connect(pydm_slider.channelValueChanged)
+    signals.new_value_signal.emit(new_channel_value)
+
+    assert pydm_slider.value_label.text() == pydm_slider.format_string.format(pydm_slider.value)
+    if not is_slider_down:
+        expected_slider_value = np.argmin(abs(pydm_slider._slider_position_to_value_map - float(new_channel_value)))
+        assert pydm_slider._slider.value() == expected_slider_value
+    else:
+        assert pydm_slider._slider.value() == 0
+
```

Comment:
A potential issue I always worry about with the slider is that the channel's value will change via some external means, which causes the slider to update, and somehow the slider's position change causes the widget to change the value again.  It would be nice to hook up a slot to the slider's `send_value_signal` that writes to `new_channel_value`, and at the end of the test, ensure that `new_channel_value` has not changed.

Reply:
I think I've already addressed this concern with the test_internal_slider_value_changed() test. @mattgibbs, could you take another look?